homeassistant/components/wemo/light.py

```diff
@@ -150,45 +152,59 @@ def available(self):
 
     def turn_on(self, **kwargs):
         """Turn the light on."""
-        transitiontime = int(kwargs.get(ATTR_TRANSITION, 0))
+        try:
+            transitiontime = int(kwargs.get(ATTR_TRANSITION, 0))
 
-        hs_color = kwargs.get(ATTR_HS_COLOR)
+            hs_color = kwargs.get(ATTR_HS_COLOR)
 
-        if hs_color is not None:
-            xy_color = color_util.color_hs_to_xy(*hs_color)
-            self.wemo.set_color(xy_color, transition=transitiontime)
+            if hs_color is not None:
+                xy_color = color_util.color_hs_to_xy(*hs_color)
+                self.wemo.set_color(xy_color, transition=transitiontime)
 
-        if ATTR_COLOR_TEMP in kwargs:
-            colortemp = kwargs[ATTR_COLOR_TEMP]
-            self.wemo.set_temperature(mireds=colortemp, transition=transitiontime)
+            if ATTR_COLOR_TEMP in kwargs:
+                colortemp = kwargs[ATTR_COLOR_TEMP]
+                self.wemo.set_temperature(mireds=colortemp, transition=transitiontime)
 
-        if ATTR_BRIGHTNESS in kwargs:
-            brightness = kwargs.get(ATTR_BRIGHTNESS, self.brightness or 255)
-            self.wemo.turn_on(level=brightness, transition=transitiontime)
-        else:
-            self.wemo.turn_on(transition=transitiontime)
+            if ATTR_BRIGHTNESS in kwargs:
+                brightness = kwargs.get(ATTR_BRIGHTNESS, self.brightness or 255)
+                self.wemo.turn_on(level=brightness, transition=transitiontime)
```

Comment:
Make a local dictionary with the parameters and populate the dictionary depending on the service data kwargs. Then make a single call to turn on using the dictionary as kwargs. Don't include the turn on call in the if else block.